Source/JavaScriptCore/runtime/JSONObject.cpp

```diff
@@ -627,6 +637,517 @@ bool Stringifier::Holder::appendNextProperty(Stringifier& stringifier, StringBui
+            recordFailure("non-standard object prototype"_s);
+            return;
+        }
+        if (!m_checkedObjectPrototype) {
+            if (mayHaveToJSON(*m_globalObject.objectPrototype())) {
+                recordFailure("object prototype may have toJSON"_s);
+                return;
+            }
+            m_checkedObjectPrototype = true;
+        }
+        if (!remainingCapacity()) {
+            recordBufferFull();
+            return;
+        }
+        m_buffer[m_length++] = '{';
+        if (canPerformFastPropertyEnumerationForJSONStringify(&structure)) {
+            structure.forEachProperty(m_vm, [&](const PropertyTableEntry& entry) -> bool {
+                if (entry.attributes() & PropertyAttribute::DontEnum)
+                    return true;
+                auto& name = *entry.key();
+                if (name.isSymbol()) {
+                    recordFailure("symbol"_s);
+                    return false;
+                }
+                if (!name.is8Bit()) {
+                    recordFailure("16-bit property name"_s);
+                    return false;
+                }
+                bool needComma = m_buffer[m_length - 1] != '{';
+                unsigned nameLength = name.length();
+                if (remainingCapacity() < needComma + 1 + nameLength + 2) {
+                    recordBufferFull();
+                    return false;
+                }
+                if (needComma)
+                    m_buffer[m_length++] = ',';
+                m_buffer[m_length] = '"';
+                auto* characters = name.characters8();
+                for (unsigned i = 0; i < nameLength; ++i) {
+                    auto character = characters[i];
+                    if (WTF::escapedFormsForJSON[character]) {
+                        recordFailure("property name character needs escaping"_s);
+                        return false;
+                    }
+                    m_buffer[m_length + 1 + i] = character;
+                }
+                m_buffer[m_length + 1 + nameLength] = '"';
+                m_buffer[m_length + 1 + nameLength + 1] = ':';
+                m_length += 1 + nameLength + 2;
+                append(object.getDirect(entry.offset()));
+                return !haveFailure();
+            });
+            if (haveFailure())
+                return;
+        } else {
+#if !FAST_STRINGIFY_GETTER_EXPERIMENT
+            recordFastPropertyEnumerationFailure(object);
+            return;
+#else
+            auto scope = DECLARE_THROW_SCOPE(m_vm);
+            PropertyNameArray names(m_vm, PropertyNameMode::Strings, PrivateSymbolMode::Exclude);
+            object.methodTable()->getOwnPropertyNames(&object, &m_globalObject, names, DontEnumPropertiesMode::Exclude);
+            if (UNLIKELY(scope.exception())) {
+                scope.clearException();
+                recordFailure("getOwnPropertyNames exception"_s);
+                return;
+            }
```

Comment:
I think we need to keep this error. It is throwing an error and it is observable because `getOwnPropertyNames` can have side-effect. So, we need to throw this error (exact this error) to users.
So, let's not clear exception, and capture it in the root caller of `FastStringifier`, and just throwing an error.